src/main/launcher/bs-config.c

```diff
@@ -0,0 +1,271 @@
+    "lte/enable", 0)
+PSMAP_OPTIONAL(PSMAP_TYPE_STR,  struct bootstrap_startup_config, lte_config_file,
+    "lte/file", "/dev/nvram/lte-config.xml")
+PSMAP_OPTIONAL(PSMAP_TYPE_STR,  struct bootstrap_startup_config, lte_unlimited_key,
+    "lte/unlimited_key", "")
+
+PSMAP_OPTIONAL(PSMAP_TYPE_STR,  struct bootstrap_startup_config, ssl_options,
+    "ssl/option", "")
+
+PSMAP_OPTIONAL(PSMAP_TYPE_BOOL, struct bootstrap_startup_config, esign_enable,
+    "esign/enable", 0)
+
+PSMAP_OPTIONAL(PSMAP_TYPE_BOOL, struct bootstrap_startup_config, eamuse_enable,
+    "eamuse/enable", 1)
+PSMAP_OPTIONAL(PSMAP_TYPE_BOOL, struct bootstrap_startup_config, eamuse_sync,
+    "eamuse/sync", 1)
+PSMAP_OPTIONAL(PSMAP_TYPE_BOOL, struct bootstrap_startup_config, eamuse_enable_model,
+    "eamuse/enable_model", 0)
+PSMAP_OPTIONAL(PSMAP_TYPE_STR,  struct bootstrap_startup_config, eamuse_config_file,
+    "eamuse/file", "/dev/nvram/ea3-config.xml")
+PSMAP_OPTIONAL(PSMAP_TYPE_BOOL, struct bootstrap_startup_config, eamuse_updatecert_enable,
+    "eamuse/updatecert_enable", 1)
+PSMAP_OPTIONAL(PSMAP_TYPE_U32,  struct bootstrap_startup_config, eamuse_updatecert_interval,
+    "eamuse/updatecert_interval", 0)
+PSMAP_END
+
+PSMAP_BEGIN(bootstrap_psmap)
+PSMAP_REQUIRED(PSMAP_TYPE_STR, struct bootstrap_config, release_code, "/release_code")
+PSMAP_END
+// clang-format on
+
+#define ROOT_NODE "/config"
+#define MODULE_PATH_PREFIX "modules/"
+
+const char *const inherited_nodes[] = {
+    "develop",
+    "default",
+    "log",
+    "minidump",
+    "boot",
+    "drm",
+    "ssl",
+    "eamuse",
+    "shield",
+    "esign",
+    "dongle",
+    "lte",
+};
+
+void bootstrap_config_init(struct bootstrap_config *config)
+{
+    memset(config, 0, sizeof(*config));
+}
+
+bool bootstrap_config_from_property(
+    struct bootstrap_config *config,
+    struct property *prop,
+    const char *selector)
+{
+    struct property_node *bootstrap_config =
+        property_search(prop, NULL, ROOT_NODE);
+    if (!bootstrap_config) {
+        log_warning(ROOT_NODE ": missing");
+        return false;
+    }
+    log_misc(ROOT_NODE ": loading...");
+    if (!property_psmap_import(
+            NULL, bootstrap_config, config, bootstrap_psmap)) {
+        log_warning(ROOT_NODE ": load failed");
+        return false;
+    }
+
+    /* Setup root startup node */
+    struct property_node *startup_root =
+        property_search(NULL, bootstrap_config, "startup");
+    if (!startup_root) {
+        log_warning(ROOT_NODE "/startup: missing");
+        return false;
+    }
+    struct property_node *startup_config =
+        property_search(NULL, startup_root, selector);
+    if (!startup_config) {
+        log_warning(ROOT_NODE "/startup/%s: missing", selector);
+        return false;
+    }
+
+    /* Resolve inheritance */
+    struct property_node *startup_parent = startup_config;
+    for (;;) {
+        char inherit_name[64];
```

Comment:
Are you sure that buffer is large enough?

Reply:
This is the same buffer size that official bootstrap uses.